### ISSUE TITLE
why zero_grad twice?

### ISSUE THREAD

**Issue by YifeiLin0226** [bhuvanakundumani/SimCSE_unsupervised#1]
Hi, thank you for the implementation, may I ask why you choose to call zero_grad() twice in training:
```
for step, batch in enumerate(tqdm(train_dataloader, desc="Iteration")):
  #for input_ids,attention_mask,token_type_ids in train_dataloader:
    batch = tuple(t.to(device) for t in batch)
    input_ids,attention_mask,token_type_ids = batch
    #import ipdb; ipdb.set_trace();
    # zero the parameter gradients
    optimizer.zero_grad()
    outputs = model(input_ids,attention_mask,token_type_ids)
    loss = outputs["loss"]

    if args.gradient_accumulation_steps > 1:
      loss = loss / args.gradient_accumulation_steps
    loss.backward()
    running_loss += loss.item()
    if (step + 1) % args.gradient_accumulation_steps == 0:
      optimizer.step()
      scheduler.step()  # Update learning rate schedule
      model.zero_grad()
```

In each iteration over the data loader, wouldn't calling optimizer.zero_grad() clear the previous gradients that might not have been used for the update yet?
